native/vendor/google/billing/build-params/BillingFlowParams.h

```diff
@@ -0,0 +1,171 @@
+/****************************************************************************
+ Copyright (c) 2024 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#pragma once
+
+#include <vector>
+
+#include "base/Macros.h"
+#include "base/RefCounted.h"
+#include "vendor/google/billing/result-values/ProductDetails.h"
+
+namespace cc {
+class CC_DLL BillingFlowParams {
+public:
+    class SubscriptionUpdateParams {
+    public:
+        class Builder {
+        public:
+            Builder& setOldPurcchaseToken(const std::string& purchaseToken) {
+                this->_purchaseToken = purchaseToken;
+                return *this;
+            }
+            Builder& setOriginalExternalTransactionId(const std::string& externalTransactionId) {
+                this->_externalTransactionId = externalTransactionId;
+                return *this;
+            }
+            Builder& setSubscriptionReplacementMode(int subscriptionReplacementMode) {
+                this->_subscriptionReplacementMode = subscriptionReplacementMode;
+                return *this;
+            }
+            SubscriptionUpdateParams* build() {
+                return new SubscriptionUpdateParams(_subscriptionReplacementMode, _purchaseToken, _externalTransactionId);
+            }
+
+        private:
+            int _subscriptionReplacementMode;
+            std::string _purchaseToken;
+            std::string _externalTransactionId;
+        };
+        static Builder* newBuilder() {
+            return new Builder();
+        }
+
+    private:
+        SubscriptionUpdateParams(int subscriptionReplacementMode, const std::string& purchaseToken, const std::string& externalTransactionId) {

```

Comment:
Could use rvalue reference to avoid `std::string` memory copy operation.

Reply:
Same as other `Params` and ·Builder·.

Reply:
fixed.